### PR TITLE
Issue #SB-24324 fix: Portal backend error log

### PR DESCRIPTION
**Pull request by kamaleeswary**

### CODE REVIEW

File: src/app/helpers/telemetryHelper.js [hunk 1]
```diff
@@ -529,5 +529,23 @@ module.exports = {
       type: 'UserSession'
     });
     return cdata;
+  },
+
+/**
+ * This function used to generate api_access log event
+ */
+  getTelemetryAPISuceessData: function(result, req, uri) {
```
Comment (vinukumar-vs) [inline src/app/helpers/telemetryHelper.js:537]: We dont need new method here. We can use the existing log method itself by passing right EDATA.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: src/app/helpers/telemetryHelper.js (hunk 1 above)
Comment (kamaleeswary) [reply src/app/helpers/telemetryHelper.js:537]: in this method, I m preparing the req data for the existing function.

File: src/app/libs/sb_telemetry_util/telemetryService.js [hunk 2]
```diff
@@ -320,22 +327,21 @@ telemetryService.prototype.generateApiCallLogEvent = function (data) {
...
   try {
-  if ( (data.responseCode !== 'OK' && data.responseCode !== 200) || res.statusCode !== 200) {
-  const result = data.params;
-  if (result) {
-  const edata = {
-    err: data.responseCode  || res.statusCode,
-    errtype: result.err || res.statusMessage,
-    requestid:  result.resmsgid || 'null',
-    errmsg: result.errmsg  || 'null',
+    const result = data.params;
```
[the remark this answers is not in the document]
Comment (kamaleeswary) [reply src/app/libs/sb_telemetry_util/telemetryService.js:330]: we don't need this 'if ( (data.responseCode !== 'OK' && data.responseCode !== 200) || res.statusCode !== 200) '
check here, only the error events should call this function then there is no need to check the statuscode. the rest inside the data i just changed the param name to match the standard that's it.

File: src/app/routes/groupRoutes.js [hunk 3]
```diff
@@ -42,26 +42,29 @@ function proxyObj() {
...
             } catch (err) {
-                const uri = 'learner/group'
-                const context = {
-                    env: telemtryEventConfig.URL[uri].env || 'group'
-                }
-                const option = telemetry.getTelemetryAPIError(JSON.parse(resData),proxyRes, context);
-                if (option) {
-                telemetryHelper.logApiErrorEventV2(req, option);
-                }
-                logger.error({ msg: 'learner route : userResDecorator json parse error:', proxyResData })
```
[the remark this answers is not in the document]
Comment (kamaleeswary) [reply src/app/routes/groupRoutes.js:64]: catch block has only the code or any functional failure right. if any API event is failed, then it will be in try block only, and there we have one more condition right to check the status…if we don’t use the condition also it will not go to catch block right?

**Review by balakrishna10 — APPROVED** [review]

File: src/app/helpers/telemetryHelper.js [hunk 4]
```diff
@@ -131,17 +131,18 @@ module.exports = {
...
       { 'category': apiConfig && apiConfig.category },
-      { 'url': apiConfig && apiConfig.url },
+      { 'url': options.path || apiConfig && apiConfig.url },
       { 'duration': Date.now() - new Date(options.headers.ts) },
       { 'status': statusCode },
       { 'protocol': 'https' },
       { 'method': options.method },
-      { 'req': options.body }
+      { 'traceid': traceid },
+      { 'req': options.body },
```
[the remark this answers is not in the document]
Comment (kamaleeswary) [reply src/app/helpers/telemetryHelper.js:145]: this is req.body i didn't add...its already defined in this method...

File: src/app/helpers/telemetryHelper.js [hunk 5]
```diff
@@ -285,20 +289,30 @@ module.exports = {
       })
     }
   },
-  logApiErrorEventV2: function (req, options) {
+  logApiErrorEventV2: function (req, options, exception = false) {
```
[the remark this answers is not in the document]
Comment (kamaleeswary) [reply src/app/helpers/telemetryHelper.js:292]: ok

File: src/app/helpers/telemetryHelper.js [hunk 6]
```diff
@@ -131,17 +131,17 @@ module.exports = {
...
       { 'title': apiConfig && apiConfig.title },
       { 'category': apiConfig && apiConfig.category },
-      { 'url': apiConfig && apiConfig.url },
+      { 'url': options.path || apiConfig && apiConfig.url },
       { 'duration': Date.now() - new Date(options.headers.ts) },
       { 'status': statusCode },
       { 'protocol': 'https' },
       { 'method': options.method },
-      { 'req': options.body }
+      { 'traceid': traceid }
```
Comment (VenkatKadiveti) [inline src/app/helpers/telemetryHelper.js:144]: is req not required here ?
why did you remove ?

File: src/app/helpers/telemetryHelper.js [hunk 7]
```diff
@@ -217,15 +217,18 @@ module.exports = {
...
     let object = {}
     let params = []
-    if (req.options) {
-      params = this.getParamsData(req.options, req.statusCode, req.resp, req.uri)
+    if (req.reqObj) {
+      params = this.getParamsData(req.reqObj, req.statusCode, req.resp, uri)
     }
-    const edata = telemetry.logEventData('api_access', 'INFO', apiConfig.message, params)
+
+    const edata = telemetry.logEventData('api_access', 'INFO', apiConfig.message,JSON.stringify(params))
```
Comment (VenkatKadiveti) [inline src/app/helpers/telemetryHelper.js:230]: use constants instead of using hard coded strings

File: src/app/helpers/telemetryHelper.js [hunk 8]
```diff
@@ -54,7 +54,7 @@ module.exports = {
     context.did = req.session.deviceId
     context.rollup = telemetry.getRollUpData(dims)
     const actor = telemetry.getActorData(req.session.userId, 'user')
-   console.log('logging session start event', context.did);
+    console.log('logging session start event', context.did);
```
Comment (VenkatKadiveti) [inline src/app/helpers/telemetryHelper.js:57]: we have logger instead of using console.log we can use 
logger.info({ msg: 'string' });

**Review by balakrishna10 — APPROVED** [review]

**Review by balakrishna10 — COMMENTED** [review]
Please check Venkats review.

**Review by balakrishna10 — COMMENTED** [review]
Please check Venkats review.

File: src/app/helpers/telemetryHelper.js (hunk 8 above)
Comment (kamaleeswary) [reply src/app/helpers/telemetryHelper.js:57]: this one I haven't add, this line is already there, only lint errors are fixed.

File: src/app/helpers/telemetryHelper.js (hunk 6 above)
Comment (kamaleeswary) [reply src/app/helpers/telemetryHelper.js:144]: this method is for entry log , in the previous review vinu said , we no need to pass the req body in this method.so I removed it here

File: src/app/helpers/telemetryHelper.js (hunk 7 above)
Comment (kamaleeswary) [reply src/app/helpers/telemetryHelper.js:230]: ok

**Review by VenkatKadiveti — APPROVED** [review]

**Review by vinukumar-vs — APPROVED** [review]
